cloudformation/fsx-substack.cfn.json

```diff
@@ -0,0 +1,257 @@
+                {
+                  "Ref": "FSXOptions"
+                }
+              ]
+            },
+            "NONE"
+          ]
+        }
+      ]
+    }
+  },
+  "Outputs": {
+    "FileSystemId": {
+      "Description": "ID of the FileSystem",
+      "Value": {
+        "Fn::If": [
+          "CreateFSX",
+          {
+            "Ref": "FileSystem"
+          },
+          {
+            "Fn::Select": [
+              "1",
+              {
+                "Ref": "FSXOptions"
+              }
+            ]
+          }
+        ]
+      }
+    }
+  },
+  "Parameters": {
+    "ComputeSecurityGroup": {
+      "Description": "SecurityGroup for FSx filesystem",
+      "Type": "String"
+    },
+    "FSXOptions": {
+      "Description": "Comma separated list of efs related options, 4 parameters in total",
+      "Type": "CommaDelimitedList"
+    },
+    "SubnetId": {
+      "Description": "SubnetId for FSx filesystem",
+      "Type": "String"
+    }
+  },
+  "Resources": {
+    "FileSystem": {
+      "Type": "AWS::FSx::FileSystem",
+      "Properties": {
+        "FileSystemType": "LUSTRE",
+        "KmsKeyId": {
+          "Fn::If": [
+            "UseFSXKMSKey",
+            {
+              "Fn::Select": [
+                "3",
+                {
+                  "Ref": "FSXOptions"
+                }
+              ]
+            },
+            {
+              "Ref": "AWS::NoValue"
+            }
+          ]
+        },
+        "LustreConfiguration": {
+          "ImportPath": {
+            "Fn::If": [
+              "UseImportPath",
+              {
+                "Fn::Select": [
+                  "5",
+                  {
+                    "Ref": "FSXOptions"
+                  }
+                ]
+              },
+              {
+                "Ref": "AWS::NoValue"
+              }
+            ]
+          },
+          "WeeklyMaintenanceStartTime": {
+            "Fn::If": [
+              "UseWeeklyMaintenanceStartTime",
+              {
+                "Fn::Select": [
+                  "6",
+                  {
+                    "Ref": "FSXOptions"
+                  }
+                ]
+              },
+              {
+                "Ref": "AWS::NoValue"
+              }
+            ]
+          },
+          "ImportedFileChunkSize": {
+            "Fn::If": [
+              "UseImportedFileChunkSize",
+              {
+                "Fn::Select": [
+                  "4",
+                  {
+                    "Ref": "FSXOptions"
+                  }
+                ]
+              },
+              {
+                "Ref": "AWS::NoValue"
+              }
+            ]
+          }
+        },
+        "StorageCapacity": {
+          "Fn::If": [
+            "UseStorageCap",
+            {
+              "Fn::Select": [
+                "2",
+                {
+                  "Ref": "FSXOptions"
+                }
+              ]
+            },
+            3600
```

Comment:
We can also have AWS::NoValue and let CloudFormation set the default

Reply:
Good idea 👍🏻